src/main/java/org/opentestsystem/delivery/logging/EventLogger.java

```diff
@@ -7,31 +7,149 @@
+    log(formatMessage(logEvent, message), getFieldMap(app, logEvent, checkpoint, sessionId, data));
+  }
+
+  /**
+   * Log event errors to centralized logging.
+   * <p>
+   * These events are intended to be sent to a logstash server and searchable using elastic search.
+   *
+   * @param app  name of the app writing the log
+   * @param logEvent  name of the event occurring
+   * @param checkpoint  an optional checkpoint name
+   * @param sessionId  an optional unique identifier common to a set of log entries (generally a request's session ID)
+   * @param message  an optional free-format message string to add to the log entry
+   * @param data optional searchable event data to add to the recorded event
+   */
+  public static void error(final String app, final ILogEvent logEvent, final String checkpoint, final String sessionId,
+                           String message, final Map<IEventData, Object> data, final Exception e) {
+    error(formatMessage(logEvent, message), getFieldMap(app, logEvent, checkpoint, sessionId, data), e);
+  }
+
+  private static void log(final String event, final Map<String, Object> fields) {
+    try {
+      logger.info(marker, format("EVENT:%s JSON:({\"event_data\": %s})", event, writer.writeValueAsString(fields)));
+    } catch (Exception e) {
+      logger.warn(marker, format("exception occurred while logging event: %s", event), e);
+    }
+  }
+
+  private static void error(final String event, final Map<String, Object> fields, final Exception e) {
+    try {
+      logger.error(marker, format("EVENT:%s JSON:({\"event_data\": %s})", event, writer.writeValueAsString(fields)),
+        e);
+    } catch (Exception inner) {
+      logger.error(marker, format("exception occurred while error event: %s", event), inner);
+    }
+  }
+
+  private static void addTimerMetrics(final ILogEvent event, final String checkpoint, Map<IEventData, Object> fields) {
+    Long eventTime = eventTimes.get(event);
+    switch (checkpoint) {
+      case "ENTER":
+        // entry point - (re)set eventTime and emit nothing
+        eventTimes.put(event, System.currentTimeMillis());
+        break;
+      case "EXIT":
+        // exit point - remove timestamp. falls through to default to emit metric.
+        eventTimes.remove(event);
+        // no break;
+      default:
+        // non entry/exit or missing position marker - emit elapsed metric
+        if (null != eventTime) {
+          fields.put(EventData.ELAPSED_TIME, System.currentTimeMillis() - eventTime);
         }
+        break;
+    }
+  }
+
+  private static Map<String, Object> getFieldMap(final String app, final ILogEvent event, String checkpoint,
+                                                 final String sessionId, final Map<IEventData, Object> data) {
+    Map<IEventData, Object> fields = new HashMap<>();
+    fields.put(EventData.APP, app);
+    if (null != checkpoint && !checkpoint.isEmpty()) {
```

Comment:
We've been using apache commons `StringUtils.isEmpty()` and `StringUtils.isNotEmpty()` for these checks